### PR TITLE
feat: add CheckSingleAgree, CheckRoomAgree, ListPermitUser, GetGroupChat

### PR DESCRIPTION
**Pull request by w3xse7en**
CheckSingleAgree: https://work.weixin.qq.com/api/doc/90000/90135/91782
CheckRoomAgree: https://work.weixin.qq.com/api/doc/90000/90135/91782
ListPermitUser: https://work.weixin.qq.com/api/doc/90000/90135/91614
GetGroupChat: https://work.weixin.qq.com/api/doc/90000/90135/92951

### CODE REVIEW

File: msg_audit.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		RoomID: roomId,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.AgreeInfo, nil
+}
+
+// ListPermitUser 获取会话内容存档开启成员列表
+func (c *WorkwxApp) ListPermitUser(Type uint8) ([]string, error) {
```
Comment (w3xse7en) [inline msg_audit.go:26]: `type` is tencent origin name, should we rename another name?

Comment (xen0n) [reply msg_audit.go:26]: According to upstream docs, that parameter is actually "level of the purchased plan" or "flavor of the purchased service"? If that's the case, the parameter definitely have to be renamed. Something like `serviceFlavor` should be OK.

And you can always add to the documentation. A clear parameter name and a Chinese explanation would be best.

Comment (w3xse7en) [reply msg_audit.go:26]:
```
// 会话内容存档版本
type MsgAuditVersion uint8

const (
	// 会话内容存档办公版
	MsgAuditVersionOffice MsgAuditVersion = 1
	// 会话内容存档服务版
	MsgAuditVersionService MsgAuditVersion = 2
	// 会话内容存档企业版
	MsgAuditVersionEnterprise MsgAuditVersion = 3
)
```
what about `MsgAuditVersion` ? i think that better than `serviceFlavor`, `MsgAudit` reference from tencent name

Comment (xen0n) [reply msg_audit.go:26]: Ah the prefixing sounds good, the methods are not namespaced (they all live on `WorkwxApp`) so having a prefix does make things stand out. But the "版本" part doesn't directly translate to "version" in the commonly used sense (some form of a thing at a particular instant of time). What about "variant" then?

BTW please prefix the docstrings with the identifier name -- this is mandated by godoc convention and is checked by linters. If this repo's CI doesn't check for them, I'll most certainly enable the check and fix things before 1.0 release.

Comment (w3xse7en) [reply msg_audit.go:26]:
```
// MsgAuditEdition 会话内容存档版本
type MsgAuditEdition uint8

const (
	// MsgAuditEditionOffice 会话内容存档办公版
	MsgAuditEditionOffice MsgAuditEdition = 1
	// MsgAuditEditionService 会话内容存档服务版
	MsgAuditEditionService MsgAuditEdition = 2
	// MsgAuditEditionEnterprise 会话内容存档企业版
	MsgAuditEditionEnterprise MsgAuditEdition = 3
)
```

`Edition` reference from msdn windows10 like `Home Edition` `Education Edition` `Enterprise Edition` and so on

Comment (xen0n) [reply msg_audit.go:26]: Yes `edition` should also work. The constants should have the whole type name prefixed though, like `MsgAuditEditionOffice` to match typical Go coding conventions.

File: models.go [hunk 2]
```diff
@@ -586,3 +586,130 @@ type JSCodeSession struct {
...
+type CheckSingleAgreeUserInfo struct {
+	UserID         string `json:"userid"`
+	ExternalOpenID string `json:"exteranalopenid"`
+}
+
+// CheckSingleAgreeInfo 获取会话同意情况（单聊）同意信息
+type CheckSingleAgreeInfo struct {
+	StatusChangeTime int    `json:"status_change_time"`
+	AgreeStatus      string `json:"agree_status"`
+	CheckSingleAgreeUserInfo
```
Comment (xen0n) [inline models.go:632]: put this at top of struct body

File: models.go [hunk 3]
```diff
@@ -586,3 +586,130 @@ type JSCodeSession struct {
...
+
+type respListPermitUser struct {
+	respCommon
+	IDs []string `json:"ids"`
+}
+
+// CheckSingleAgreeUserInfo 获取会话同意情况（单聊）内外成员
+type CheckSingleAgreeUserInfo struct {
+	UserID         string `json:"userid"`
+	ExternalOpenID string `json:"exteranalopenid"`
```
Comment (xen0n) [inline models.go:625]: Ahh this typo is on Tencent's side and we can do nothing about this 😂

File: models.go [hunk 4]
```diff
@@ -586,3 +586,130 @@ type JSCodeSession struct {
...
+	MemberID int `json:"memberid"`
+	JoinTime int `json:"jointime"`
+}
+
+// GroupChat 获取会话内容存档内部群信息
+type GroupChat struct {
+	Members        []GroupChatMember `json:"members"`
+	RoomName       string            `json:"roomname"`
+	Creator        string            `json:"creator"`
+	RoomCreateTime int               `json:"room_create_time"`
```
Comment (xen0n) [inline models.go:693]: expose this and other timestamp fields as `time.Time` for ergonomic downstream usage? see `media_api.go` for one example of doing the conversion before returning to biz logic.

**Comment by xen0n** [comment]
LGTM from a cursory look, if this works for you then I have no problem including this.

Thanks for the addition!

bors r+

(P.S. the API surface is getting big, Tencent is steadily adding ever more function blocks over the years and it is not going to slow down anytime soon, the whole ecosystem is established and thriving. I'm considering breaking blocks of features into separate minor types. I'll send a PR over the coming weeks detailing the change and you can review!)

**Pull request merged**